src/redux/movieDetailsSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { RootState } from "./store";
+
+const initialState = {
+  movieDetailsStatus: false,
+};
+
+const movieDetailsSlice = createSlice({
+  name: "movieDetails",
+  initialState,
+  reducers: {
+    openMovieDetails(state) {
+      state.movieDetailsStatus = true;
+    },
+    closeMovieDetails(state) {
+      state.movieDetailsStatus = false;
+    },
+  },
+});
+
+export const selectMovieDetailsStatus = (state: RootState) =>
+  state.movieDetails.movieDetailsStatus;
+
+export const { openMovieDetails, closeMovieDetails } =
+  movieDetailsSlice.actions;
+
+export default movieDetailsSlice.reducer;
```

Comment:
You don't need it at all, create an action that will save movie details
If you click on card save movie's data and open `Movie details` if there is movie's data
click on search icon clean store and that will close `Movie details`